src/Compilers/VisualBasic/BasicAnalyzerDriver/VisualBasicDeclarationComputer.vb

```diff
@@ -61,36 +61,52 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     For Each decl In t.Members
                         ComputeDeclarationsCore(model, decl, shouldSkip, getSymbol, builder, newLevel, cancellationToken)
                     Next
-                    builder.Add(GetDeclarationInfo(model, node, getSymbol, cancellationToken))
+                    Dim attributes = GetAttributes(t.EnumStatement.AttributeLists)
+                    builder.Add(GetDeclarationInfo(model, node, getSymbol, attributes, cancellationToken))
+                    Return
+                Case SyntaxKind.EnumStatement
+                    Dim t = CType(node, EnumStatementSyntax)
+                    Dim attributes = GetAttributes(t.AttributeLists)
+                    builder.Add(GetDeclarationInfo(model, node, getSymbol, attributes, cancellationToken))
                     Return
                 Case SyntaxKind.EnumMemberDeclaration
                     Dim t = CType(node, EnumMemberDeclarationSyntax)
-                    builder.Add(GetDeclarationInfo(model, node, getSymbol, t.Initializer, cancellationToken))
+                    Dim attributes = GetAttributes(t.AttributeLists)
+                    Dim codeBlocks = SpecializedCollections.SingletonEnumerable(Of SyntaxNode)(t.Initializer).Concat(attributes)
+                    builder.Add(GetDeclarationInfo(model, node, getSymbol, codeBlocks, cancellationToken))
                     Return
                 Case SyntaxKind.DelegateSubStatement, SyntaxKind.DelegateFunctionStatement
                     Dim t = CType(node, DelegateStatementSyntax)
                     Dim paramInitializers As IEnumerable(Of SyntaxNode) = GetParameterInitializers(t.ParameterList)
-                    builder.Add(GetDeclarationInfo(model, node, getSymbol, paramInitializers, cancellationToken))
+                    Dim attributes = GetAttributes(t.AttributeLists)
+                    Dim codeBlocks = paramInitializers.Concat(attributes)
+                    builder.Add(GetDeclarationInfo(model, node, getSymbol, codeBlocks, cancellationToken))
                     Return
                 Case SyntaxKind.EventBlock
                     Dim t = CType(node, EventBlockSyntax)
                     For Each decl In t.Accessors
                         ComputeDeclarationsCore(model, decl, shouldSkip, getSymbol, builder, newLevel, cancellationToken)
                     Next
                     Dim eventInitializers = GetParameterInitializers(t.EventStatement.ParameterList)
-                    builder.Add(GetDeclarationInfo(model, node, getSymbol, eventInitializers, cancellationToken))
+                    Dim attributes = GetAttributes(t.EventStatement.AttributeLists)
```

Comment:
>Dim attributes = GetAttributes(t.EventStatement.AttributeLists) [](start = 20, length = 63)

Aren't we doing this for the event statement below?

Reply:
This function can be called with either EventBlockSyntax or EventStatementSyntax, but we cannot have both paths taken in same call.

---
In reply to: [152418061](https://github.com/dotnet/roslyn/pull/23318#discussion_r152418061) [](ancestors = 152418061)

Reply:
[https://github.com/dotnet/roslyn/issues/23347](https://github.com/dotnet/roslyn/issues/23347) to ensure we have clarity here.

---
In reply to: [152421256](https://github.com/dotnet/roslyn/pull/23318#discussion_r152421256) [](ancestors = 152421256,152418061)